### PR TITLE
SRX-XYKGST Don't deploy queued version when unlocking

### PR DESCRIPTION
**Pull request by mouhsen-ibrahim**
* rename Queue Lock Behavior to Record
* Remove "Delete Queue" button and action from UI
* Stop deploying queued version when deleting a lock

### CODE REVIEW

File: services/cd-service/pkg/repository/repository.go [hunk 1]
```diff
@@ -1098,30 +1097,6 @@ func (s *State) ProcessQueue(ctx context.Context, fs billy.Filesystem, environme
 			// whenever the next deployment happens:
 			err = s.DeleteQueuedVersion(environment, application)
 			return fmt.Sprintf("deleted queued version %d because it was already deployed. app=%q env=%q", *queuedVersion, application, environment), err
```
Comment (tameremad) [inline services/cd-service/pkg/repository/repository.go:1099]: Why not remove the entire `ProcessQueue` function or make it `return "", nil`?

File: services/cd-service/pkg/repository/transformer_test.go [hunk 2]
```diff
@@ -1458,26 +1459,22 @@ func TestTransformer(t *testing.T) {
...
 					t.Fatal(err)
 				}
-				if i == nil {
-					t.Errorf("unexpected version: expected 1, actual nil")
-				} else {
-					if *i != 1 {
-						t.Errorf("unexpected version: expected 1, actual %d", *i)
-					}
+				if i != nil {
+					t.Errorf("unexpected version: unexpected nil")
```
Comment (tameremad) [inline services/cd-service/pkg/repository/transformer_test.go:1470]: Can you explain the error message, please? 

```suggestion
					t.Errorf("unexpected version %d: expected: nil", i)
```

File: services/cd-service/pkg/repository/transformer_test.go [hunk 3]
```diff
@@ -1458,26 +1459,22 @@ func TestTransformer(t *testing.T) {
...
 				}
 				q, err := s.GetQueuedVersion("production", "test")
 				if err != nil {
 					t.Fatal(err)
 				}
-				if q != nil {
-					t.Errorf("unexpected version: expected nil, actual %d", *q)
+				if q == nil {
+					t.Errorf("unexpected version: expected %d, actual nil", *q)
```
Comment (tameremad) [inline services/cd-service/pkg/repository/transformer_test.go:1477]: I think the expected value here is 1. 
```suggestion
					t.Errorf("unexpected version: expected 1, actual nil")
```

File: services/frontend-service/src/legacy-ui/ReleaseDialog.tsx [hunk 4]
```diff
@@ -366,17 +352,6 @@ const ReleaseEnvironment: VFC<{
...
-            deleteQueue: {
-                application: applicationName,
-                environment: environmentName,
-                currentlyDeployedVersion: currentlyDeployedVersion,
-            },
-        }),
-        [applicationName, environmentName, currentlyDeployedVersion]
-    );
-
     const queuedVersion = overview.environments[environmentName].applications[applicationName]?.queuedVersion || 0;
```
Comment (tameremad) [inline services/frontend-service/src/legacy-ui/ReleaseDialog.tsx:355]: Question: Should we remove the queuedVersion for here as well?

Comment (mouhsen-ibrahim) [reply services/frontend-service/src/legacy-ui/ReleaseDialog.tsx:355]: The story only included removing the button and not the message

File: services/cd-service/pkg/repository/repository.go (hunk 1 above)
Comment (mouhsen-ibrahim) [reply services/cd-service/pkg/repository/repository.go:1099]: We still need it to delete the queue if necessary

**Review by tameremad — APPROVED** [review]
:+1:

**Pull request merged**